src/StripeEmitter.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @private
+ * (For internal use in stripe-node.)
+ * Wrapper around the Event Web API.
+ */
+class _StripeEvent extends Event {
+  data?: RequestEvent | ResponseEvent;
+  constructor(eventName: string, data: any) {
+    super(eventName);
+    this.data = data;
+  }
+}
+
+/** Minimal EventEmitter wrapper around EventTarget. */
+class StripeEmitter {
+  eventTarget: EventTarget;
+  listenerMapping: Record<string, (...args: any[]) => any>;
```

Comment:
Instead of being a `Record<string` should it be a `Map<object, ...`? Calling toString on function objects seems a bit hacky.

Reply:
This is discouraged by TypeScript, see the following linting error:
```
Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys.eslint[@typescript-eslint/ban-types](https://github.com/typescript-eslint/typescript-eslint/blob/v4.33.0/packages/eslint-plugin/docs/rules/ban-types.md)
```

According to [this SO post](https://stackoverflow.com/a/10858667), it looks like everything gets turned into a string anyways under the hood, so I'd rather be explicit here.

Reply:
https://stackoverflow.com/questions/10858632/are-functions-valid-keys-for-javascript-object-properties talks about object properties, I was wondering if we should use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map 
>Any value (both objects and [primitive values](https://developer.mozilla.org/en-US/docs/Glossary/Primitive)) may be used as either a key or a value

Reply:
```
const makeFunc = () => {
    const time = Date.now()
    return () => console.log(time);
}
const a = makeFunc();
const b = makeFunc();

const dict: Record<string, any> = {};
dict[a.toString()] = a;
dict[b.toString()] = b;

Object.values(dict).forEach(f=>f());
```
illustrates the problem with using function.toString()

Reply:
Ah sorry, I misunderstood! Thanks for the example, `Map` is definitely what we want. :)